uxarray/grid/grid.py

```diff
@@ -1393,3 +1399,39 @@ def isel(self, **dim_kwargs):
             raise ValueError(
                 "Indexing must be along a grid dimension: ('n_node', 'n_edge', 'n_face')"
             )
+
+    def get_dual(self, method="global"):
+        """Compute the dual mesh for a grid, returns a new grid object.
+
+         Parameters
+        ----------
+        method: str, default="global"
+            Method for constructing the dual mesh, either "global" or "local"
+
+        Returns
+        --------
+        dual : Grid
+            Dual Mesh Grid constructed
+        """
+
+        if method == "local":
+            raise ValueError("Local Dual Mesh is not yet supported, use global")
+        elif method != "global":
+            raise ValueError(
+                f"Invalid method: {method}. Please use a supported method instead"
+            )
+
+        if _check_duplicate_nodes_indices(self):
+            raise RuntimeError(
+                "Duplicate nodes found, consider using `Grid.merge_duplicate_node_indices()`"
+            )
+
+        # Get dual mesh node face connectivity
+        dual_node_face_conn = construct_dual(grid=self)
+
```

Comment:
This is where we would check for duplicate nodes and set any full fill value row to match it's duplicate.